dhis-2/dhis-api/src/main/java/org/hisp/dhis/analytics/event/DimensionWrapper.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) 2004-2021, University of Oslo
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.analytics.event;
+
+import java.util.Date;
+
+import org.hisp.dhis.common.BaseDimensionalItemObject;
+import org.hisp.dhis.common.BaseDimensionalObject;
+import org.hisp.dhis.common.BaseNameableObject;
+import org.hisp.dhis.common.ValueTypedDimensionalItemObject;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class DimensionWrapper
+{
```

Comment:
I think we could use `DimensionalItemObject` and `BaseDimensionalItemObject` here, as they have the same properties/methods as this class.

Reply:
I don't like this class as well. 
I'm working on a twin ticket for enrollments and I'm planning to remove this class.
However the problem is in the way FE wants the response:

- `valueType`, when possible (i.e. for all `ValueTypedDimensionalObjects`)
- `dimensionType` should hold `baseDimensionalItemObject.dimensionItemType` and `baseDimensionalObject.dimensionType`, dependingly

now not all `BaseDimensionalItemObjects` are also `ValueTypedDimensionalItemObjects`, and `dimensionItemType!=dimensionType`

so all in all, we need a specific definition for FE response.

So what I'm doing in my next ticket is returning a collection of `BaseIdentifiableObjects` in the services, and then mapping in an FE-specific DTO to accommodate the response, each type with its own mapper.

Reply:
Okay understand, thanks. I understand that e.g. DataElement is a dimension _item_ in aggregate analytics, and _dimension_ for event analytics. DataElement does not implement DimensionalObject, so we might need a "wrapper".